src/Vault.sol

```diff
@@ -13,6 +13,90 @@ import { PrizePool } from "v5-prize-pool/PrizePool.sol";
 import { TwabController } from "v5-twab-controller/TwabController.sol";
 import { Claimer } from "v5-vrgda-claimer/Claimer.sol";
 
+/// @notice Emitted when the TWAB controller is set to the zero address
+error TwabControllerZeroAddress();
+
+/// @notice Emitted when the Yield Vault is set to the zero address
+error YieldVaultZeroAddress();
+
+/// @notice Emitted when the Prize Pool is set to the zero address
+error PrizePoolZeroAddress();
+
+/// @notice Emitted when the Owner is set to the zero address
+error OwnerZeroAddress();
+
+/// @notice Emitted when the amount to deposit is greater than the max available for the receiver
```

Comment:
I'm not sure `available` describes correctly the behavior.
Based on the EIP, I would suggest `max amount allowed`: https://eips.ethereum.org/EIPS/eip-4626#maxdeposit
```suggestion
/// @notice Emitted when the amount being deposited for the receiver is greater than the max amount allowed
```